fuse_examples/classification/multimodality/model_tabular_imaging.py

```diff
@@ -0,0 +1,295 @@
+                 imaging_backbone: torch.nn.Module=None,
+                 multimodal_backbone: torch.nn.Module=None,
+                 tabular_heads: Sequence[torch.nn.Module]=None,
+                 imaging_heads: Sequence[torch.nn.Module]=None,
+                 multimodal_heads: Sequence[torch.nn.Module]=None,
+                 ) -> None:
+        super().__init__()
+
+        self.tabular_inputs = tabular_inputs
+        self.tabular_backbone = tabular_backbone
+        if self.tabular_backbone:
+            self.add_module('tabular_backbone', self.tabular_backbone)
+
+        self.imaging_inputs = imaging_inputs
+        self.imaging_backbone = imaging_backbone
+        if self.imaging_backbone:
+            self.add_module('imaging_backbone', self.imaging_backbone)
+
+        self.multimodal_backbone = multimodal_backbone
+        if self.multimodal_backbone:
+            self.add_module('multimodal_backbone', multimodal_backbone)
+
+
+        self.tabular_heads = torch.nn.ModuleList(tabular_heads)
+        if self.tabular_heads:
+            self.add_module('tabular_heads', self.tabular_heads)
+
+        self.imaging_heads = torch.nn.ModuleList(imaging_heads)
+        if self.imaging_heads:
+            self.add_module('imaging_heads', self.imaging_heads)
+
+        self.multimodal_heads = torch.nn.ModuleList(multimodal_heads)
+        if self.multimodal_heads:
+            self.add_module('multimodal_heads', self.multimodal_heads)
+
+    def tabular_modules(self):
+        return [self.tabular_backbone, self.tabular_heads]
+
+    def imaging_modules(self):
+        return [self.imaging_backbone, self.imaging_heads]
+
+    def multimodal_modules(self):
+        return [self.multimodal_backbone, self.multimodal_heads]
+
+    def forward(self, batch_dict: Dict) -> Dict:
+
+        if self.tabular_backbone:
+            tabular_features = self.tabular_backbone.forward(batch_dict)
+
+        if self.imaging_backbone:
+            imaging_input = FuseUtilsHierarchicalDict.get(batch_dict, self.imaging_inputs[0][0])
+            imaging_features = self.imaging_backbone.forward(imaging_input)
+            FuseUtilsHierarchicalDict.set(batch_dict, 'model.imaging_features', imaging_features)
+
+        if self.multimodal_backbone:
+            multimodal_features = self.multimodal_backbone.forward(batch_dict)
+            FuseUtilsHierarchicalDict.set(batch_dict, 'model.multimodal_features', multimodal_features)
+
+
+        # run through heads
+        if self.tabular_heads:
+            for head in self.tabular_heads:
+                batch_dict = head.forward(batch_dict)
+
+        if self.imaging_heads:
+            for head in self.imaging_heads:
+                batch_dict = head.forward(batch_dict)
+
+        if self.multimodal_heads:
+            for head in self.multimodal_heads:
+                batch_dict = head.forward(batch_dict)
+
+        return batch_dict['model']
+
+
+if __name__ == '__main__':
```

Comment:
remove. add it as a unitest if it's important

Reply:
will remove at the end of the review